modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/data_set/filters/modals/selection_filter/source_type/ApiRestApplication.tsx

```diff
@@ -341,7 +340,9 @@ function ApiRestApplication({
 				}
 			});
 		}
-	}, [selectedRESTApplication, selectedRESTSchema]);
+
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [selectedRESTSchema]);
```

Comment:
This is a bit scary, as it leads to stale state. Probably the only acceptable way to have this lint exception is when you want something done only once, when there are no dependencies in the list.

What is the purpose of removing this dependency? Could it be achieved in some other way?

Reply:
The thing here is the hierarchy of data. To select a new rest application leads to have a new rest schema, but we need to calculate fields for a different schemas under the same rest application. Not removing the dependency of `selectedRESTApplication` leads to use the wrong `selectedRESTSchema` for fields fetching.

One way to remove the useEffect/dependency stuff is to invoke an util in every place `selectedRESTSchema` changes but, personally, I prefer having it as side effects decoupled from UI stuff.

Reply:
> To select a new rest application leads to have a new rest schema, but we need to calculate fields for a different schemas under the same rest application.

Well, yes. The issue is, removing `selectedRESTApplication` from the list of dependencies will results in no re-calculation on selection of REST application. `getFields` within `useEffect` will not trigger. This will lead to stale `fields` state, matching the previously selected REST application.

Reply:
What I would highly recommend doing here is removing `useEffect`, like you mentioned above.